tests/koa-request.spec.js

```diff
@@ -31,6 +31,21 @@ describe('Koa Escher Authentication Middleware suite', function() {
       .expect(401, 'Test escher error', done);
   });
 
+  it('should return with HTTP 400 in case of application error', function(done) {
```

Comment:
Just a minor change, please change the text of the test to: "should return with the original error in case of application error". It seems that it converts all application error to 400.